main.go

```diff
@@ -27,6 +29,9 @@ func main() {
 	if *namespace != "" {
 		log.Printf("watching namespace %s.", *namespace)
 	}
+	if *namespacePrefix != "" {
```

Comment:
It may be worth it to add some validation for the `namespace-prefix`. We may want to check whether it starts with `/` or if it ends with `/`. In the case of the latter, we may want to remove the trailing `/` and assume, throughout the code, that the prefix doesn't end with it.

As it is now I can pass `secrets/test` as my namespace prefix and that will fail the `pathAllowed` check.

This valudation would help improve the user experience

Reply:
Vault has no idea what `/` is, its just a part of the key name and doesn't mean it is a directory or anything like that. For this particular feature we expect that keys can only be accessed under `$prefix$namespace/$key`. We shouldn't be preventing a user from using whatever prefix they prefer like `secret/cluster_name-` and writing keys as `secret/cluster_name-namespace/key`.

> As it is now I can pass secrets/test as my namespace prefix and that will fail the pathAllowed check.

It doesn't fail if that really is your intended prefix. Both of these test cases pass and are valid:

```
		{
			name:      "prefixes can be anything",
			path:      "secret/testnamespace/key",
			prefix:    "secret/test",
			namespace: "namespace",
			want:      true,
		},
		{
			name:      "prefixes can be anything",
			path:      "secret/test-namespace/key",
			prefix:    "secret/test-",
			namespace: "namespace",
			want:      true,
		},
```

Reply:
> For this particular feature we expect that keys can only be accessed under $prefix$namespace/$key.

This is what I was missing and now that you mention this expectation (specifically the format), I feel like it's not entirely clear in this patch. By reading the documentation, it feels the expectation is for this prefix to end with a `/` and that the expectation is that it'll be a 'path like` prefix.

:stop_sign: Could you expand the documentation to express this expectation and, perhaps, add a couple of tests for cases that don't end with a `/`? I'd like to make sure users know that `kube-vault-controller` makes no assumption about what the prefix format is and that whatever is passed to the CLI param will be prefixed to the namespace.

TBH, I'm not a fan of this lack of expectation. It feels like, by not enforcing a format, we're either allowing users to shoot themselves on the foot or missing an opportunity to bring consistency into a deployment. I understand that vault has no idea what `/` is but `/` has become the common separator used to create a tree-like structure in vault's key-value storage. All this said the fact that `vault` should be the one responsible for any enforcement is stronger than my wish for consistency.

Reply:
Updated the docs and tests to try to clarify things. Does it look better now?